File: chapter1/binary_search.rb
def binary_search(arr, low, high, target)
  if low > high
    return -1;
  end

  mid = (low + high) / 2

  if target > arr[mid]
    binary_search arr, mid+1, high, target
  elsif target < arr[mid]
    binary_search arr, low, mid-1, target
  else
    arr[mid]
  end

end

File: chapter1/binarySearch.js
function binarySearch(arr, low, high, target) {
    if (low > high) {
        return -1;
    }

    let mid = (low + high) / 2;
    if (target > arr[mid]) {
        return binarySearch(arr, mid + 1, high, target);
    } else if (target < arr[mid]) {
        return binarySearch(arr, low, high - 1, target);
    } else {
        return arr[mid];
    }
}

 module.exports = binarySearch;